File: benchmark/cases/node_fr_16_ecs_donnees/workspace/Monde.js
class Monde {
    constructor() {
        this.entites = new Map(); // ID -> Set de composants
        this.prochainID = 1;
    }

    creerEntite() {
        const id = this.prochainID++;
        this.entites.set(id, new Map());
        return id;
    }

    ajouterComposant(entiteID, composant) {
        const entite = this.entites.get(entiteID);
        if (entite) {
            entite.set(composant.constructor.name, composant);
        }
    }

    obtenirEntitesAvec(nomsComposants) {
        const resultats = [];
        for (const [id, composants] of this.entites) {
            if (nomsComposants.every(nom => composants.has(nom))) {
                resultats.push({ id, composants });
            }
        }
        return resultats;
    }
}

module.exports = Monde;

File: benchmark/cases/python_fr_13_ecs_bus_evenements/workspace/monde.py
class Monde:
    def __init__(self, bus_evenements):
        self.entites = {}
        self.bus_evenements = bus_evenements
        self.prochain_id = 1

    def creer_entite(self):
        eid = self.prochain_id
        self.prochain_id += 1
        self.entites[eid] = {}
        return eid

    def ajouter_composant(self, eid, composant):
        type_comp = type(composant).__name__
        self.entites[eid][type_comp] = composant
